fdbserver/ptxn/test/RealStorageServer.actor.cpp

```diff
@@ -0,0 +1,146 @@
+
+	KeyValueStoreType storeType = KeyValueStoreType::SSD_BTREE_V2;
+	state std::string folder = ".";
+	std::string fileName = joinPath(folder, "storage-" + ssi->id().toString() + "." + storeType.toString());
+	std::cout << "new Storage Server file name: " << fileName << std::endl;
+	deleteFile(fileName);
+	state IKeyValueStore* data = openKVStore(storeType, fileName, ssi->id(), 0);
+
+	state ReplyPromise<InitializeStorageReply> storageReady;
+
+	// Initialize dbInfo with mocked cursor and mocked loy system.
+
+	// TODO: Use Reference API?
+	// TODO: delete mockLogSystem eventually.
+	state MockLogSystem* mockLogSystem = new MockLogSystem();
+	mockLogSystem->cursor = mockPeekCursor.castTo<ILogSystem::IPeekCursor>();
+
+	LogSystemConfig logSystemConfig;
+	logSystemConfig.logSystemType = LogSystemType::mock;
+	logSystemConfig.mockLogSystem = mockLogSystem;
+
+	ServerDBInfo dbInfoBuilder;
+	dbInfoBuilder.recoveryState = RecoveryState::ACCEPTING_COMMITS;
+	dbInfoBuilder.logSystemConfig = logSystemConfig;
+	state Reference<AsyncVar<ServerDBInfo>> dbInfo = makeReference<AsyncVar<ServerDBInfo>>(dbInfoBuilder);
+	// Initialize dbInfo done.
+
+	std::cout << "Starting Storage Server" << std::endl;
+	state Future<Void> ss = storageServer(data, *ssi, tag, storageReady, dbInfo, folder);
+	actors->emplace_back(ss);
+
+	// Delay 0.1 seconds to wait storage server processing the mutations.
+	// TODO: Wait until certain desired version is reached rather than predefined delay time.
+	wait(delay(0.1));
+	std::cout << "After waiting for 0.1 seconds" << std::endl;
+
+	return Void();
+}
+
+ACTOR Future<Void> driverVerifyGetValue(StorageServerInterface* ssi,
+                                        KeyRef key,
+                                        Version version,
+                                        ValueRef expectedValue) {
+	GetValueRequest getValueRequest(UID(), // spanContext
+	                                key,
+	                                version,
+	                                Optional<TagSet>(), // tags
+	                                Optional<UID>()); // debugID
+	getValueRequest._requestTime = timer();
+
+	ssi->getValue.send(getValueRequest);
+
+	GetValueReply getValueReply = wait(getValueRequest.reply.getFuture());
+
+	const Value& value = getValueReply.value.get();
+	std::cout << "Get value: " << value.toString() << std::endl;
+	ASSERT(value == expectedValue);
+
+	return Void();
+}
+
+struct StorageServerTestDriver {
+	Reference<MockPeekCursor> mockPeekCursor;
+
+	// Default tag.
+	Tag tag = Tag(1, 1);
+	// Set by initEndpoints()
+	StorageServerInterface ssi;
+	std::vector<Future<Void>> actors;
+
+	StorageServerTestDriver() { initEndpoints(); }
+
+	void initEndpoints() {
+		UID uid = nondeterministicRandom()->randomUniqueID();
+		ssi = StorageServerInterface(uid);
+		ssi.locality = LocalityData();
+		ssi.initEndpoints();
+	}
+
+	Future<Void> runStorageServer() { return driverRunStorageServer(&ssi, &actors, tag, mockPeekCursor); }
```

Comment:
FYI, an alternative to have separate `driverRunStorageServer` actor is to declare a static actor member, e.g.,
```
ACTOR static Future<Void> runStorageServer(StorageServerInterface* ssi,
                                          std::vector<Future<Void>>* actors,
                                          Tag tag,
                                          Reference<MockPeekCursor> mockPeekCursor,
                                          StorageServerTestDriver* self) { ... }
```